hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/cmd/ExecuteOperationCommand.java

```diff
@@ -19,68 +19,61 @@
 import java.util.Collection;
 import java.util.Map;
 
-import org.hawkular.agent.monitor.extension.MonitorServiceConfiguration;
 import org.hawkular.agent.monitor.inventory.ID;
-import org.hawkular.agent.monitor.inventory.InventoryIdUtil;
-import org.hawkular.agent.monitor.inventory.InventoryIdUtil.ResourceIdParts;
 import org.hawkular.agent.monitor.inventory.ManagedServer;
-import org.hawkular.agent.monitor.inventory.Name;
 import org.hawkular.agent.monitor.inventory.ResourceManager;
 import org.hawkular.agent.monitor.inventory.dmr.DMRInventoryManager;
 import org.hawkular.agent.monitor.inventory.dmr.DMROperation;
 import org.hawkular.agent.monitor.inventory.dmr.DMRResource;
-import org.hawkular.agent.monitor.inventory.dmr.LocalDMRManagedServer;
-import org.hawkular.agent.monitor.inventory.dmr.RemoteDMRManagedServer;
 import org.hawkular.agent.monitor.log.AgentLoggers;
 import org.hawkular.agent.monitor.log.MsgLogger;
 import org.hawkular.bus.common.BasicMessageWithExtraData;
 import org.hawkular.cmdgw.api.ExecuteOperationRequest;
 import org.hawkular.cmdgw.api.ExecuteOperationResponse;
-import org.hawkular.cmdgw.api.MessageUtils;
-import org.hawkular.cmdgw.api.ResponseStatus;
+import org.hawkular.dmr.api.OperationBuilder;
 import org.hawkular.dmrclient.Address;
-import org.hawkular.dmrclient.CoreJBossASClient;
-import org.hawkular.dmrclient.JBossASClient;
 import org.hawkular.inventory.api.model.CanonicalPath;
 import org.jboss.as.controller.client.ModelControllerClient;
-import org.jboss.dmr.ModelNode;
 
 /**
  * Execute an operation on a resource.
  */
-public class ExecuteOperationCommand implements Command<ExecuteOperationRequest, ExecuteOperationResponse> {
+public class ExecuteOperationCommand extends
+        AbstractResourcePathCommand<ExecuteOperationRequest, ExecuteOperationResponse> {
     private static final MsgLogger log = AgentLoggers.getLogger(ExecuteOperationCommand.class);
     public static final Class<ExecuteOperationRequest> REQUEST_CLASS = ExecuteOperationRequest.class;
 
+    public ExecuteOperationCommand() {
+        super("Execute Operation", "DMR Node");
+    }
+
+    /** @see org.hawkular.agent.monitor.cmd.AbstractResourcePathCommand#createResponse() */
     @Override
-    public BasicMessageWithExtraData<ExecuteOperationResponse> execute(
-            BasicMessageWithExtraData<ExecuteOperationRequest> envelope, CommandContext context) throws Exception {
-        ExecuteOperationRequest request = envelope.getBasicMessage();
-        log.infof("Received request to execute operation [%s] on resource [%s]", request.getOperationName(),
```

Comment:
> I still want to see in the log message that a request was received to execute operation O on resource R.

Hm... yes, the message we log in https://github.com/hawkular/hawkular-agent/blob/master/hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/cmd/AbstractResourcePathCommand.java#L76 does not contain the name of the Operation.

> Or maybe we could make these attributes not final to change the [Execute Operation] and [DMR Node] on runtime.

@josejulio , yeah, that would solve the problem, but I'd personally vote for keeping those final so that there is no state stored outside the `execute()` method. It is true, that we actually do not have such a contract here (whereas we have it in `command-gateway`), but immutability is still worth keeping because it makes the design cleaner and perhaps also because it allows JIT compiler to do more aggressive optimalizations. I see two options how to keep the GenericCommand and its parent immutable and get more informative log and error messages:

(a) add a `getOperationName` method to `AbstractResourcePathCommand` like this
```
    protected String getOperationName(BasicMessageWithExtraData<REQ> envelope) {
        return operationName;
    }
```
and use the value it returns in error and log messages. 

In `ExecuteOperationCommand`, it could simply be overriden:
```
    protected String getOperationName(BasicMessageWithExtraData<ExecuteOperationRequest> envelope) {
        return envelope.getBasicMessage().getOperationName();
    }
```

(b) Do not make `ExecuteOperationCommand` a subclass of `AbstractResourcePathCommand`

I tend to prefer (a)

WDYT?